tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -249,4 +249,88 @@ object SQLPlanParser extends Logging {
     maxDuration
   }
 
+  def parseFilterExpressions(exprStr: String): Array[String] = {
+    val result = ArrayBuffer[String]()
+
+    // Filter ((isnotnull(s_state#68) AND (s_state#68 = TN)) OR (hex(cast(value#0 as bigint)) = B))
+    // split on AND/OR/NOT
+    val exprSepAND = if (exprStr.contains("AND")) {
+      exprStr.split(" AND ").map(_.trim)
+    } else {
+      Array(exprStr)
+    }
+    val exprSepOR = if (exprStr.contains(" OR ")) {
+      exprSepAND.flatMap(_.split(" OR ").map(_.trim))
+    } else {
+      exprSepAND
+    }
+
+    val exprSplit = if (exprStr.contains("NOT ")) {
+      result += "Not"
+      exprSepOR.flatMap(_.split("NOT ").map(_.trim))
+    } else {
+      exprSepOR
+    }
+
+    // Remove paranthesis from the beginning and end to get only the expressions
+    val paranRemoved = exprSplit.map(_.replaceAll("""^\(+""", "").replaceAll("""\)\)$""", ")"))
+
+    val functionPattern = """(\w+)\(.*\)""".r
+    val conditionalExprPattern = """([(\w# )]+) ([+=<>|]+) ([(\w# )]+)""".r
+
+    paranRemoved.foreach { case expr =>
+      if (expr.contains(" ")) {
+        // likely some conditional expression
+        // TODO - add in arithmetic stuff (- / * )
+        conditionalExprPattern.findFirstMatchIn(expr) match {
+          case Some(func) =>
+            logWarning(s" found expr: $func")
+            if (func.groupCount < 3) {
+              logError(s"found incomplete expression - $func")
+            }
+            val lhs = func.group(1)
+            functionPattern.findFirstMatchIn(lhs) match {
```

Comment:
I think this should be the same logic as the else below. make common function

Reply:
created common function